### PR TITLE
New: timestamp unit for startAt

### PR DESCRIPTION
**Pull request by DanDeMicco**
Adds support for timestamp parsing for startAt. A timestamp is a youtube style timestamp such as 5h2m1s and will start media previews at that point.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @DanDeMicco has signed the CLA. Thanks for the pull request!

File: src/lib/viewers/media/MediaBaseViewer.js [hunk 1]
```diff
@@ -787,6 +791,56 @@ class MediaBaseViewer extends BaseViewer {
...
     }
+
+    /**
+     * Converts from a youtube style timestamp to seconds
+     *
+     * @param {string} timestamp - the youtube style timestamp eg. "1h2m3s"
+     * @return {number} - the time in seconds
+     */
+    convertTimestampToSeconds(timestamp) {
+        let timeInSeconds = 0;
```
Comment (jeremypress) [inline src/lib/viewers/media/MediaBaseViewer.js]: use INITIAL_TIME_IN_SECOND?

File: src/lib/viewers/media/MediaBaseViewer.js [hunk 2]
```diff
@@ -787,6 +791,56 @@ class MediaBaseViewer extends BaseViewer {
...
+        const minutes = MINUTE_REGEX.exec(timestamp);
+        const seconds = SECOND_REGEX.exec(timestamp);
+
+        /**
+         * Validates a substring match and converts to float
+         *
+         * @param {string} match - the timestamp substring e.g. 1h, 2m, or 3s
+         * @return {number} - the number for the given unit
+         */
+        const getValueOfMatch = (match) => {
```
Comment (jeremypress) [inline src/lib/viewers/media/MediaBaseViewer.js]: Might want to break this out for testing purposes

File: src/lib/viewers/media/MediaBaseViewer.js [hunk 3]
```diff
@@ -787,6 +791,56 @@ class MediaBaseViewer extends BaseViewer {
...
+            // Strip off unit (h, m, s) and convert to float
+            const parsedMatch = parseFloat(match[0].slice(0, -1), 10);
+            return Number.isNaN(parsedMatch) ? 0 : parsedMatch;
+        };
+
+        if (hours) {
```
Comment (jeremypress) [inline src/lib/viewers/media/MediaBaseViewer.js]: very readable! 👍

**Pull request merged**